h/db/__init__.py

```diff
@@ -44,31 +45,24 @@
 Session = sessionmaker()
 
 
-def init(engine, base=Base, should_create=False, should_drop=False, authority=None):
-    """Initialise the database tables managed by `h.db`."""
-    # Import models package to populate the metadata
-    import h.models  # pylint: disable=unused-import
-
-    if should_drop:  # pragma: no cover
-        # SQLAlchemy doesn't know about the report schema, and will end up
-        # trying to drop tables without cascade that have dependent tables
-        # in the report schema and failing. Clear it out first.
-        engine.execute("DROP SCHEMA IF EXISTS report CASCADE")
-        base.metadata.reflect(engine)
-        base.metadata.drop_all(engine)
-    if should_create:  # pragma: no cover
-        # In order to be able to generate UUIDs, we load the uuid-ossp
-        # extension.
-        engine.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
-        base.metadata.create_all(engine)
+def pre_delete(engine):  # pragma: no cover
+    engine.execute("DROP SCHEMA IF EXISTS report CASCADE")
+
+
+def pre_create(engine):  # pragma: no cover
+    engine.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
+
+
+def post_create(engine):  # pragma: no cover
+    authority = environ["AUTHORITY"]
 
     default_org = _maybe_create_default_organization(engine, authority)
     _maybe_create_world_group(engine, authority, default_org)
 
 
-def make_engine(settings):  # pragma: no cover
+def create_engine(database_url):  # pragma: no cover
```

Comment:
`make_engine()` is renamed to `create_engine()` for consistency with our other apps. Also it takes the `database_url()` directly rather than taking the entire Pyramid `settings` dict so it can be called from a standalone script outside of any Pyramid context.